pkg/kubefed2/disable.go

```diff
@@ -252,6 +284,15 @@ func deleteFederatedTypeConfig(client genericclient.Client, typeConfig *fedv1a1.
 	return nil
 }
 
+func generatedFederatedTypeConfig(config *rest.Config, enableTypeDirective *enable.EnableTypeDirective) (*fedv1a1.FederatedTypeConfig, error) {
+	apiResource, err := enable.LookupAPIResource(config, enableTypeDirective.Name, enableTypeDirective.Spec.TargetVersion)
+	if err != nil {
+		return nil, err
+	}
+	typeConfig := enable.GenerateTypeConfigForTarget(*apiResource, enableTypeDirective).(*fedv1a1.FederatedTypeConfig)
```

Comment:
This works. But as caller functions only needs `typeconfig.Interface`, it does not needs such type conversion here.

Reply:
This type assertion is necessary because this function returns a `*fedv1a1.FederatedTypeConfig`. It returns this because the `DisableFederation` caller is re-using its `typeConfig` variable (for simplicity) that is defined as a `*fedv1a1.FederatedTypeConfig` [here](https://github.com/kubernetes-sigs/federation-v2/pull/769/files#diff-fc79e5fb1808a541a91f4994f5f02254R187). `typeConfig` is defined this way because it is passed to multiple functions that expect different interface implementations e.g. `runtime.Object`, `typeconfig.Interface`.

Reply:
I'm fine with the code anyway. 
I guess it is compromise for 2 case:
 - use a common variable in caller function and has type conversion in the function
 - use a different temparary variable in caller function and has no type conversion internally
Outter logic makes more sense than inner logic for code reading.

Reply:
It is a compromise as you say. I prefer this implementation for a couple reasons:
1. It provides the most flexibility to the caller because it returns the same type passed in so that the caller is free to choose what interface implementation to use without performing any type assertion.
2. If I used a temporary variable in `DisableFederation`, I'd have to make 2 different calls to `deleteFederatedType` using 2 different `typeConfig` variables. Instead, this approach simplifies the code in `DisableFederation`.

I am happy to change it if you and others feel strongly.

Reply:
I think its fine the way it is. Type conversion is anyways needed at some place, because of what @font mentioned [here](https://github.com/kubernetes-sigs/federation-v2/pull/769/commits/2f9112de2089e8f6b58415c7686b91df784d6cc8#r276788968). Also I think it makes sense to have this function return the concrete type as the caller can receive it in either the concrete type or an interface.
Also, let this not lead to bikeshedding.. :)